kotlin-inject-compiler/ksp/src/main/kotlin/me/tatarka/inject/compiler/ksp/Util.kt

```diff
@@ -35,7 +35,7 @@ private fun KSAnnotation.hasName(packageName: String, simpleName: String): Boole
     // we can skip resolving if the short name doesn't match
     if (shortName.asString() != simpleName) return false
     val declaration = annotationType.resolve().declaration
-    return declaration.packageName.asString() == packageName
+    return declaration.qualifiedName?.asString() == "$packageName.$simpleName"
```

Comment:
https://github.com/google/ksp/issues/1721

Reply:
the upstream bug is fixed now?

Reply:
I think I tried it earlier and it still had issues, but that might have been conflated with the other KSP2 issues. I'll try again.

Reply:
I reverted the commit so it could be unreverted in case it is still needed.